server/datastore/mysql/labels.go

```diff
@@ -444,29 +446,44 @@ func (ds *Datastore) DeleteLabel(ctx context.Context, name string) error {
 			if err == sql.ErrNoRows {
 				return ctxerr.Wrap(ctx, notFound("Label").WithName(name))
 			}
-			return ctxerr.Wrapf(ctx, err, "getting label id to delete")
+			return ctxerr.Wrap(ctx, err, "getting label id to delete")
 		}
-
-		_, err = tx.ExecContext(ctx, `DELETE FROM labels WHERE id = ?`, labelID)
-		if err != nil {
+		if err := deleteLabelsInTx(ctx, tx, []uint{labelID}); err != nil {
 			if isMySQLForeignKey(err) {
 				return ctxerr.Wrap(ctx, foreignKey("labels", name), "delete label")
 			}
-			return ctxerr.Wrapf(ctx, err, "delete label")
+			return ctxerr.Wrap(ctx, err, "delete labels in tx")
 		}
+		return nil
+	})
+}
 
-		_, err = tx.ExecContext(ctx, `DELETE FROM label_membership WHERE label_id = ?`, labelID)
-		if err != nil {
-			return ctxerr.Wrapf(ctx, err, "delete label_membership")
-		}
+func deleteLabelsInTx(ctx context.Context, tx sqlx.ExtContext, labelIDs []uint) error {
+	query, args, err := sqlx.In(`DELETE FROM labels WHERE id IN (?)`, labelIDs)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "build sqlx.In statement for labels")
+	}
+	if _, err = tx.ExecContext(ctx, query, args...); err != nil {
+		return ctxerr.Wrap(ctx, err, "delete label")
+	}
 
-		_, err = tx.ExecContext(ctx, `DELETE FROM pack_targets WHERE type=? AND target_id=?`, fleet.TargetLabel, labelID)
-		if err != nil {
-			return ctxerr.Wrapf(ctx, err, "deleting pack_targets for label %d", labelID)
-		}
+	query, args, err = sqlx.In(`DELETE FROM label_membership WHERE label_id IN (?)`, labelIDs)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "build sqlx.In statement for label_membership")
+	}
+	if _, err = tx.ExecContext(ctx, query, args...); err != nil {
+		return ctxerr.Wrap(ctx, err, "delete label_membership")
+	}
 
-		return nil
-	})
+	query, args, err = sqlx.In(`DELETE FROM pack_targets WHERE type = ? AND target_id IN (?)`, fleet.TargetLabel, labelIDs)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "build sqlx.In statement for pack_targets")
+	}
+	if _, err = tx.ExecContext(ctx, query, args...); err != nil {
+		return ctxerr.Wrap(ctx, err, "deleting pack_targets")
+	}
+
```

Comment:
I'm seeing various on-delete behaviors depending on which fkey we have:

- RESTRICT (installers, VPP apps, in house apps)
- SET NULL (profiles, declarations)
- CASCADE (policies, queries)

This means that currently profiles/declaration team associations will flip to null when the underlying label is deleted, while software being associated will fail the delete. Betting there's a bug in here on profiles/declarations.

Guessing the intention here is that anything where deleting the label could widen the scope of applying something should be restricted, but that means profiles/declarations/policies/queries have broken behavior here.